OpenRA.Mods.Common/UpdateRules/Rules/20180923/RemoveNegativeDamageFullHealthCheck.cs

```diff
@@ -0,0 +1,71 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class RemoveNegativeDamageFullHealthCheck : UpdateRule
+	{
+		public override string Name { get { return "Negative damage weapons are no longer unable to target actors with full health."; } }
+		public override string Description
+		{
+			get
+			{
+				return "Negative damage weapons are no longer unable to target actors with full health.\n" +
+					"To keep the previous behaviour use conditions from GrantConditionOnDamageState\n" +
+					"on Targetable trait.";
+			}
+		}
+
+		static readonly string[] DamageWarheads =
+		{
+			"TargetDamage",
+			"SpreadDamage",
+			"HealthPercentageDamage"
+		};
+
+		readonly Dictionary<string, List<string>> locations = new Dictionary<string, List<string>>();
+
+		public override IEnumerable<string> AfterUpdate(ModData modData)
+		{
+			if (locations.Any())
+				yield return "These are the negative damage weapons you have, you can make actors\n" +
+					"targetable by those grant a condition using GrantConditionOnDamageState and\n" +
```

Comment:
This sentence doesn't make sense to me. `by granting a condition ...`?
(The one below has issues as well.)

Reply:
`those` here refers to the weapons. I dunno if this can be worded better, but current sentence is gramatically and meaningly correct, i think.

Reply:
I agree with @abcdefg30. this is not entirely grammatically correct.

"by those grant" should be "by those by granting"...